File: Dsa-day160.cpp
class Solution {
public:
    int minimumBoxes(vector<int>& apple, vector<int>& capacity) {
        int total = 0;
        for (int a : apple) {
            total += a;
        }
        sort(capacity.rbegin(), capacity.rend());
        int sum = 0;
        for (int i = 0; i < capacity.size(); i++) {
            sum += capacity[i];
            if (sum >= total) {
                return i + 1;
            }
        }
        return capacity.size();
    }
};

File: Dsa-day160.py
class Solution:
    def minimumBoxes(self, apple: List[int], capacity: List[int]) -> int:
        total = sum(apple)
        capacity.sort(reverse=True)
        s = 0
        for i in range(len(capacity)):
            s += capacity[i]
            if s >= total:
                return i + 1
        return len(capacity)